app/javascript/components/applications/ApplicationPage.js

```diff
@@ -7,33 +7,38 @@ import mapProps from '~/utils/mapProps'
 import { mapApplication } from '~/components/mappers/soqlToDomain'
 import labelMapperFields from './application_details/applicationDetailsFieldsDesc'
 
-const buildActionLinkIfNecessary = (app) => {
-  if (!app.isLotterComplete && app.submissionType === 'Paper') {
-    return { title: 'Edit Application', link: appPaths.toApplicationEdit(app.id) }
-  } else {
-    return undefined
-  }
+const buildActionLinkIfNecessary = (app, fromPage) => {
+  const actions = []
+
+  if (!app.is_lottery_complete && app.application_submission_type === 'Paper')
```

Comment:
note this is using snake_case even though it's branched off of caritas

Reply:
It's ok here since it's using the application from the properties and not the shortForm fields